cmd/agent/api/agent/agent.go

```diff
@@ -278,6 +281,40 @@ func getRuntimeConfig(w http.ResponseWriter, r *http.Request) {
 	w.Write(runtimeConfig)
 }
 
+func setRuntimeConfig(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	setting := vars["setting"]
+	log.Info("Got a request to change a setting: %s", setting)
+	r.ParseForm()
+	value := html.UnescapeString(r.Form.Get("value"))
+
+	if _, ok := config.RuntimeSettings[setting]; !ok {
+		body, _ := json.Marshal(map[string]string{"error": "unknown setting"})
+		http.Error(w, string(body), 500)
+		return
+	}
+	if err := config.RuntimeSettings[setting].Set(value); err != nil {
+		body, _ := json.Marshal(map[string]string{"error": err.Error()})
+		http.Error(w, string(body), 500)
+		return
+	}
```

Comment:
I think the logic of setting a runtime settings should be part of the config package. The rest of the code should not have to be aware how runtime settings are stored for example.
Either `config.SetRuntimeSettings` or make runtimeSettings it's own package ?

Reply:
Having method to set / get runtime settings is good idea, however I think it makes sense to let this piece of code in the config package.